code/model/modelread.cpp

```diff
@@ -2860,6 +2850,23 @@ modelread_status read_model_file_no_subsys(polymodel * pm, const char* filename,
 		}
 	}
 
+	// ---------- submodel movement sanity checks ----------
+
+	for (i = 0; i < pm->n_models; i++) {
```

Comment:
Can we guarantee that no child subobject will be iterated before any of its parent subobjects?  The `do_movement_sanity_checks` function also sets the `Can_move` flag, and this guarantee is necessary for the flag to propagate down the submodel tree.

Reply:
Ah right, that definitely isn't guaranteed tmk. Was going to handle that. Recursively traversing the submodels through childrne is much more complicated, but some thing I'm quite used to by this point in pof tools lol.

Reply:
Hmm.  I was going to suggest using `model_iterate_submodel_tree`, but at this point in the model processing, the tree isn't set up yet.

Reply:
Another hmm.  This is in the subobject parsing code:
```
				parent = cfread_int(fp);
				sm->parent = parent;
				auto parent_sm = parent < 0 ? nullptr : &pm->submodel[parent];
```
I'd guess this code would break if there *wasn't* a guarantee that a parent is parsed before any of its children.

Reply:
I'm not convinced it would necessarily be obvious if things failed that way, but yes, the parsing does require that parents be ordered in the file before their children, this does not however mean that their id, their position in the array, is the same order, so that must also be guarded against.